### ISSUE TITLE
Segmentation fault happend when Run OpenAI Gym Examples

### ISSUE THREAD

**Issue by bin913** [caffe2/BlueWhale#2]
python run_rl_gym.py -g CartPole-v0 -l 0.1

args: Namespace(batch_size=128, constraint=False, discount_gamma=0.9, gpu=False, gymenv='CartPole-v0', learn_batch_num_every_iteration=100, learn_every_n_iterations=2, learning_rate=0.1, maxq_learning=True, model_id=u'new', model_type=u'DQN', number_iterations=1000, number_steps_timeout=-1, number_steps_total=1000000, optimizer=u'SGD', path='/home/bzhang/work/Git/reinforcement-learning-models/rlmodels/outputs/', render=False, save_iteration=-1, test=False, upload=False, verbosity=0)
Env gym:  CartPole-v0
Env setting: state/action type(shape): Box(4,) Discrete(2)
Model Id:  new
Model Type:  MODEL_T.DQN
Model Optimizer:  SGD
Model NN layers: [64] [u'relu']
test1 
test2
*** Aborted at 1508656367 (unix time) try "date -d @1508656367" if you are using GNU date ***
PC: @     0x7f046e6e2fb4 _ZN6caffe210MakeStringIJSsA3_cSsEEESsDpRKT_
*** SIGSEGV (@0xffffffffffffffe8) received by PID 7743 (TID 0x7f0487aed740) from PID 18446744073709551592; stack trace: ***
    @     0x7f04876e2330 (unknown)
    @     0x7f046e6e2fb4 _ZN6caffe210MakeStringIJSsA3_cSsEEESsDpRKT_
    @     0x7f046e6e314a caffe2::enforce_detail::EnforceFailMessage::get_message_and_free()
    @     0x7f046b8f92f6 caffe2::Tensor<>::dim32()
    @     0x7f04661bd18c caffe2::XavierFillOp<>::Fill()
    @     0x7f046619c685 caffe2::FillerOp<>::RunOnDevice()
    @     0x7f046e6e1895 caffe2::Operator<>::Run()
    @     0x7f046601dbb8 caffe2::SimpleNet::RunAsync()
    @     0x7f0466031882 caffe2::Workspace::RunNetOnce()
    @     0x7f046e6dbd2a _ZZN8pybind1112cpp_function10initializeIZN6caffe26python16addGlobalMethodsERNS_6moduleEEUlRKNS_5bytesEE22_bIS8_EINS_4nameENS_7siblingENS_5scopeEEEEvOT_PFT0_DpT1_EDpRKT2_ENKUlPNS_6detail15function_recordENS_6handleESR_SR_E1_clESQ_SR_SR_SR_.isra.1620
    @     0x7f046e6dbf57 _ZZN8pybind1112cpp_function10initializeIZN6caffe26python16addGlobalMethodsERNS_6moduleEEUlRKNS_5bytesEE22_bIS8_EINS_4nameENS_7siblingENS_5scopeEEEEvOT_PFT0_DpT1_EDpRKT2_ENUlPNS_6detail15function_recordENS_6handleESR_SR_E1_4_FUNESQ_SR_SR_SR_
    @     0x7f046e6e7b52 pybind11::cpp_function::dispatcher()
    @           0x52714b (unknown)
    @           0x555551 (unknown)
    @           0x525560 (unknown)
    @           0x555551 (unknown)
    @           0x525560 (unknown)
    @           0x5247ea (unknown)
    @           0x568b3a (unknown)
    @           0x4c2604 (unknown)
    @           0x4d1c5c (unknown)
    @           0x55f6db (unknown)
    @           0x5244dd (unknown)
    @           0x555551 (unknown)
    @           0x525560 (unknown)
    @           0x555551 (unknown)
    @           0x525560 (unknown)
    @           0x567d14 (unknown)
    @           0x465bf4 (unknown)
    @           0x46612d (unknown)
    @           0x466d92 (unknown)
    @     0x7f048732af45 (unknown)
Segmentation fault (core dumped)

**Comment by nishadsingh1** [comment]
Hi @bin913, thanks for reporting this and sorry for the delayed response. We've given this repo a bit of an overhaul since you commented and I can't reproduce the behavior you're reporting. Do you mind checking if it still occurs for you? And if so, can you give us your Caffe2 version and OS?

**Comment by anuj1560** [comment]
In my ubuntu18.04(bionic) ,Segmentation fault(core dumped) occurs when run openai(gym) examples. If I remove env.render line it works fine. Please help
$vi vart.py
import gym
env = gym.make('CartPole-v0')
for i_episode in range(20):
    observation = env.reset()
    for t in range(100):
        env.render()
        print(observation)
        action = env.action_space.sample()
        observation, reward, done, info = env.step(action)
        if done:
            print("Episode finished after {} timesteps".format(t+1))
            break

env.close()

$python3 cart.py 
Segmentation fault (core dumped)

Thanks
